core-rust/state-manager/src/transaction/executable_logic.rs

```diff
@@ -188,32 +188,22 @@ impl<'a, S: SubstateDatabase> TransactionLogic<S> for ConfiguredExecutable<'a> {
         match self {
             ConfiguredExecutable::GenesisFlash { flash_receipt } => flash_receipt.into(),
             ConfiguredExecutable::SystemFlash { state_updates } => {
-                let mut substate_schema_mapper =
-                    SubstateSchemaMapper::new(SystemDatabaseReader::new(store));
-                substate_schema_mapper.add_for_all_individually_updated(&state_updates);
-                let substate_system_structures = substate_schema_mapper.done();
-
-                // Sanity check that all updates are to existing nodes so that
-                // we can assure there are no new entities in the receipt
-                let reader = SystemDatabaseReader::new(store);
-                for (node_id, ..) in &state_updates.by_node {
-                    reader
-                        .get_object_info(*node_id)
-                        .expect("Substate flash is currently only supported for existing nodes.");
-                }
+                let application_events = Vec::new();
+                let system_structure =
+                    SystemStructure::resolve(store, &state_updates, &application_events);
+                let new_node_ids = collect_new_node_ids(&state_updates);
+                let state_update_summary =
+                    StateUpdateSummary::new(store, new_node_ids, &state_updates);
 
                 let commit_result = CommitResult {
                     state_updates,
-                    state_update_summary: Default::default(),
+                    state_update_summary,
```

Comment:
Thanks for fixing this!

I notice that we have actually done this properly now... I just want to triple check that the `state_update_summary` isn't used in the `LedgerTransactionReceipt` / for any of the hashes?

I think that's right, but just wanted to check 👍

Reply:
double-checked now - it becomes a part of `LocalTransactionExecution` only (and is served from API), but does not go into any hashes 👍